### DISCUSSION TITLE
Imola bug

### DISCUSSION THREAD

**Discussion by negfrequency** [theOehrly/Fast-F1#386] [General]
Ergast refers to 'Round 6' as Monaco however FastF1 refers to 'Round 6' in 2023 as Imola which did not happen. Might want to account for this somehow?

**Comment by negfrequency** [comment]
i guess this can be resolved by calling events by their string name as opposed to round id

**Reply by theOehrly** [reply]
> Ergast refers to 'Round 6' as Monaco however FastF1 refers to 'Round 6' in 2023 as Imola which did not happen. Might want to account for this somehow?

Fixed it, thanks for reporting this. This was an event schedule backend issue, so there's no need to update.

**Reply by Casper-Guo** [reply]
Just to be clear, does the fastf1 schedule no longer contains Imola?

**Reply by theOehrly** [reply]
> Just to be clear, does the fastf1 schedule no longer contains Imola?

No it does not. F1 has removed it from their timing data API and Ergast has removed it as well.

F1 has not removed it from their event listings API though, which is the one that provides the data for their schedule on formula1.com
That means that annoyingly, Imola is still listed as Round 6 there. 

With the current state, I'm not convinced what the best approach here actually is.
I could potentially add it again but marked as cancelled. The question then still remains whether it should be round 6 or round 7 in the user facing API of FastF1.
This would need an update to FastF1 though and I need to look into this more to be able to assess how complicated this would be.

Edit: it would need to be Round 6 probably, because else I cannot maintain compatibility between the schedule backends, given that Ergast does not know it even exists. I can only maybe add it to the schedule but without a round number.

**Reply by Casper-Guo** [reply]
I see. I have the following code that runs automatically every Monday to pull and transform data. I didn't set up error handling before last week and Imola broke it. If Imola is removed entirely and the Monaco Grand Prix round number changes, I must modify my cached data directly.

```
schedule = f.get_event_schedule(season)

loaded_rounds = set(pd.unique(existing_data["RoundNumber"]))
newest_round = schedule[schedule["EventDate"] < datetime.now()]["RoundNumber"].max()
all_rounds = set(range(1, newest_round + 1))
missing_rounds = all_rounds.difference(loaded_rounds)

for i in missing_rounds:
    race = f.get_session(2023, i, 'R')

    try:
        race.load()
    except:
        # Handles FastF1 errors
        print(f"Cannot load {race}")
```

I also need to rewrite this to use the new UTC timestamps for better consistency.

**Reply by theOehrly** [reply]
The Monaco round number has already changed. In my opinion, this is the only reasonable way that I can deal with this problem, considering that it involves multiple different APIs with different amounts of information as well as the need to maintain compatibility. 
I'm currently considering adding Imola back to the event schedule, but without a round number and with the introduction of a ``status`` flag for completed, upcoming and cancelled events. I'm not entirely sure if that is worth the hassle and whether this doesn't create too much more confusion.

**Comment by siddharth-tewari** [comment]
```Traceback (most recent call last):
  File "C:\Users\Siddharth Tewari\Desktop\Formula 1\MiniSectors\minisectors.py", line 21, in <module>
    from ollie_color_scheme import *
  File "C:\Users\Siddharth Tewari\Desktop\Formula 1\MiniSectors\ollie_color_scheme.py", line 14, in <module>
    fpath = os.path.join(rcParams["datapath"], "/Users/........./Formula1-Regular.ttf")#Download F1 font https://www.formula1.com/etc/designs/fom-website/fonts/F1Regular/Formula1-Regular.ttf
  File "C:\Users\Siddharth Tewari\AppData\Local\Programs\Python\Python39\lib\site-packages\matplotlib\__init__.py", line 679, in __getitem__
    return dict.__getitem__(self, key)
KeyError: 'datapath'
```
I got this error while trying to load data for it. Sorry, I am using it after many months so may not be knowing few new changes.

**Reply by theOehrly** [reply]
This is an error in matplotlib or rather the script you are using. The rcParam "datapath" apparently does not exist. I can't really help you with that. It's completely unrelated from FastF1.

**Reply by siddharth-tewari** [reply]
Oh ok, thank you for telling.
